atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -485,10 +483,7 @@ private static Supplier<ExecutorService> instrumentedFixedThreadPoolSupplier(
             int numberOfThriftHosts = serversConfig.numberOfThriftHosts();
             int corePoolSize = poolSize * numberOfThriftHosts;
             int maxPoolSize = maxConnectionBurstSize * numberOfThriftHosts;
-            return Tracers.wrap(MetricRegistries.instrument(
-                    registry,
-                    createThreadPoolWihtoutSpans("Atlas Cassandra KVS", corePoolSize, maxPoolSize),
-                    "Atlas Cassandra KVS"));
```

Comment:
`createThreadPoolWithoutSpans` returns an executor with both trace and metric instrumentation. We then wrap it here with trace and metric instrumentation again here.

This is a good example of why it's not ideal to have this handled implicitly in a way that is not obvious to callers.